File: Cpp_src/easy/power_of_four_342.cc
#include <iostream>
using namespace std;

// Use bit manipulation
// Time Complexity: O(1)
// Space Complexity:O(1)
// Runtime: 4 ms, faster than 56.73%
// Memory Usage: 8.1 MB, less than 80.95%
class Solution
{
public:
    bool isPowerOfFour(int num)
    {
        return (num > 0) && ((num & (num - 1)) == 0) && (num & 0x55555555) != 0;
    }
};

File: Golang_src/easy/powerOfFour_342.go
package easy

// Use Bit Manipulation:
// Time Complexity: O(1)
// Space Complexity:O(1)
// Runtime: 0 ms, faster than 100.00%
// Memory Usage: 2.1 MB, less than 100.00%
func isPowerOfFour(num int) bool {
	return num > 0 && num&(num-1) == 0 && num&0x55555555 != 0
}
